src/swarm/neo/protocol/socket/MessageGenerator.d

```diff
@@ -264,7 +260,7 @@ struct IoVecTracker
                 start = end;
             }
 
-            assert(start == this.length);
+            verify(start == this.length);
```

Comment:
With assertions that just verify the inner logic of a function I wonder if they should be converted. The guideline says
> The built-in `assert` should *only* be used as a means to spot issues earlier in the debugging process. It should never be relied upon for error checking in a live application, as it is not present in release builds.

"A means to spot issues earlier in the debugging process" reflects my intention why I put this assertion here. It is safe to remove it in a release build.

Reply:
Verbal discussion concluded:
* Our aim must be to make release builds of critical D2 programs, in order to avoid array bounds errors being thrown.
* In that case, "internal" assertions such as this one can be left as is or converted to `verify`s. It doesn't really matter.
* Generally, the current pass of conversion is intended to be simplistic. However, if someone notes a situation like this where an assertion can be left as is, there's no harm in doing so. (I don't feel we should go out of our way to look for such cases right now, though.)

Reply:
I feel this one is a matter of taste.

[The doc](https://github.com/sociomantic-tsunami/sociomantic/blob/master/Code/D/assert-vs-enforce.md#verify) says:
>`verify` is designed to be a tool for catching programmer errors and basic sanity / safety errors

which I think applies in this case too.

Reply:
Yeah, `verify` is fine here as well.